proto/proto/transaction.proto

```diff
@@ -0,0 +1,81 @@
+syntax = "proto3";
+package penumbra.transaction;
+
+// A Penumbra transaction.
+message Transaction {
+  TransactionBody body = 1;
+  // The binding signature is stored separately from the transaction body that it signs.
+  bytes binding_sig = 2;
+}
+
+// The body of a transaction.
+message TransactionBody {
+  // A list of actions (state changes) performed by this transaction.
+  repeated Action actions = 1;
+  // The root of some previous state of the Sapling note commitment tree.
+  bytes anchor = 2;
+  // The maximum height that this transaction can be included in the chain.
+  uint32 expiry_height = 3;
+  // The chain this transaction is intended for.  Including this prevents
+  // replaying a transaction on one chain onto a different chain.
+  string chain_id = 4;
+}
+
+// A state change performed by a transaction.
+message Action {
+  oneof action {
+    Fee fee = 1;
+    Spend spend = 2;
+    Output output = 3;
+  }
+}
+
+// Specifies fees paid by a transaction.
+message Fee {
+    uint64 amount = 1;
+}
+
+// Spends a shielded note.
+message Spend {
+  SpendBody body = 1;
+  // The spend authorization signature is stored separately from the spend body it authorizes.
+  bytes auth_sig = 2;
+}
+
+// The body of a spend description, stored separately from the signatures that authorize it.
+message SpendBody {
+  // A commitment to the value of the input note.
+  bytes cv = 1;
+  // The nullifier of the input note.
+  bytes nullifier = 3;
+  // The randomized validating key for the spend authorization signature.
+  bytes rk = 4;
+  // The spend proof.
+  bytes zkproof = 5;
+}
+
+// Creates a new shielded note.
+message Output {
+  OutputBody body = 1;
+  // An encrypted memo. 528 bytes.
+  bytes encrypted_memo = 2;
+  // The key material used for note encryption, wrapped in encryption to the
+  // sender's outgoing viewing key. 80 bytes.
+  bytes ovk_wrapped_key = 3;
+}
+
+// The body of an output description, not including a memo or ovk wrapping.
+// Splitting this data out allows its use where the memo / ovk wrapping is not
+// required (e.g., IBC).
```

Comment:
The plan is for IBC transfers into Penumbra to work by having the IBC Fungible Token Transfer packet on the counterparty chain specify a base64-encoded output description in the destination address field; in that case, it's better to keep the description as small as possible, and there's no clear need to have a memo field or ovk support (since the transaction is in the clear on the counterparty chain).